examples/barriers/kukas_self_collision.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2022 Domrachev Ivan,  Simeon Nedelchec
+
+"""Two iiwa14-s with full-body self-collision avoidance using hpp-fcl."""
+
+import time
+import meshcat_shapes
+import numpy as np
+import pinocchio as pin
+import qpsolvers
+from loop_rate_limiters import RateLimiter
+
+import pink
+from pink import solve_ik
+from pink.barriers import SelfCollisionBarrier
+from pink.tasks import FrameTask, PostureTask
+from pink.visualization import start_meshcat_visualizer
+
+if __name__ == "__main__":
+    # Load the robot and define the custom frames
+    urdf_path = "examples/barriers/models/iiwa14_spheres_collision.urdf"
+    srdf_path = "examples/barriers/models/iiwa14_spheres_collision.srdf"
```

Comment:
I need some help there because I'm not sure what is the best way to represent this.

The thing is, this is a copy of the `iiwa14` model from `robot_descriptions.py` with the following changes:
1. Base joint collision geometry is changed from cylinder to sphere.
2. The manipulators are doubled, and all joints and frames are renamed to `first_X` and `second_X`.
3. The `pdf` file is added (but I consider it more like a user configuration, than a part of the general-purpose model.

With that being said, I'm not sure it should be a separate entry in `robot_menagerie.py`. I've looked for other ways to create this model, the ones I've considered are:
1. Use `pin.appendModel(...)`. But I'm not sure how it handles two models with practically similar links and frame names.
2. Leave only `.urdf` and `.srdf` files in the example folder, and load all the meshes from `robot_description.py`/other repository. However, I don't know how to make it properly.

If, given the fact that this is practically a clone of `robot_menagerie.py` repository, you still think it's fine to add it there, I am 100% ready to do it. Any suggestions on other approaches to handle this are welcome too.

Reply:
I agree with @simeon-ned's suggestion to use `pin.appendModel` for the dual-arm part. Before appending you can update the collision model of the arm in Python to replace the cylinders by spheres (NB: make sure to recreate the collision data associated with the collision model after updating it).